src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -72,19 +80,27 @@ pub(crate) struct SharedBufferBatchInner {
 
 impl SharedBufferBatchInner {
     pub(crate) fn new(
+        table_id: TableId,
         epoch: HummockEpoch,
         payload: Vec<SharedBufferItem>,
         range_tombstone_list: Vec<DeleteRangeTombstone>,
         size: usize,
         _tracker: Option<MemoryTracker>,
     ) -> Self {
-        let (smallest_empty, mut smallest_table_key, mut largest_table_key, range_tombstones) =
-            Self::get_table_key_ends(range_tombstone_list);
+        let SharedBufferDeleteRangeMeta {
+            smallest_empty,
+            mut smallest_table_key,
+            mut largest_table_key,
+            range_tombstones,
+        } = Self::get_table_key_ends(table_id, range_tombstone_list);
 
         if let Some(item) = payload.last() {
-            if item.0.gt(&largest_table_key) {
-                largest_table_key.clear();
-                largest_table_key.extend_from_slice(item.0.as_ref());
+            if match &largest_table_key {
```

Comment:
In this PR we many several similar logic of comparing a key with the bound of end key. We may extract the common logic in a util method.

Reply:
> In this PR we many several similar logic of comparing a key with the bound of end key. We may extract the common logic in a util method.

fixed